packages/core/src/codewhisperer/commands/startTransformByQ.ts

```diff
@@ -194,6 +266,214 @@ export async function preTransformationUploadCode() {
+        humanInTheLoopManager.setManifestFileValues(manifestFileValues)
+
+        // 3) We need to replace version in pom.xml
+        const newPomFileVirtualFileReference = await humanInTheLoopManager.createPomFileCopy(
+            humanInTheLoopManager.getTmpDependencyListDir(),
+            pomFileVirtualFileReference
+        )
+        humanInTheLoopManager.setNewPomFileVirtualFileReference(newPomFileVirtualFileReference)
+        await humanInTheLoopManager.replacePomFileVersion(
+            newPomFileVirtualFileReference,
+            manifestFileValues.sourcePomVersion
+        )
+
+        const codeSnippet = await getCodeIssueSnippetFromPom(newPomFileVirtualFileReference)
+        // Let the user know we've entered the loop in the chat
+        transformByQState.getChatControllers()?.startHumanInTheLoopIntervention.fire({
+            tabID: ChatSessionManager.Instance.getSession().tabID,
+            codeSnippet,
+        })
+
+        // 4) We need to run maven commands on that pom.xml to get available versions
+        const compileFolderInfo = humanInTheLoopManager.getCompileDependencyListFolderInfo()
+        runMavenDependencyUpdateCommands(compileFolderInfo)
+        const xmlString = await humanInTheLoopManager.getDependencyListXmlOutput()
+        const { latestVersion, majorVersions, minorVersions, status } = await parseVersionsListFromPomFile(xmlString)
+
+        if (status === dependencyNoAvailableVersions) {
+            // let user know and early exit for human in the loop happened because no upgrade versions available
+            const error = new AlternateDependencyVersionsNotFoundError()
+
+            transformByQState.getChatControllers()?.errorThrown.fire({
+                error,
+                tabID: ChatSessionManager.Instance.getSession().tabID,
+            })
+
+            throw error
+        }
+
+        const dependencies = new DependencyVersions(
+            latestVersion,
+            majorVersions,
+            minorVersions,
+            manifestFileValues.sourcePomVersion
+        )
+
+        // 5) We need to wait for user input
+        // This is asynchronous, so we have to wait to be called to complete this loop
+        transformByQState.getChatControllers()?.promptForDependencyHIL.fire({
+            tabID: ChatSessionManager.Instance.getSession().tabID,
+            dependencies,
+        })
+    } catch (err: any) {
+        try {
+            // Regardless of the error,
+            // Continue transformation flow
+            await terminateHILEarly(jobId)
+        } finally {
+            transformByQState.getChatControllers()?.errorThrown.fire({
+                error: err,
+                tabID: ChatSessionManager.Instance.getSession().tabID,
+            })
+        }
+        CodeTransformTelemetryState.instance.setCodeTransformMetaDataField({
+            errorMessage: err.message,
+        })
+        telemetry.codeTransform_humanInTheLoop.emit({
```

Comment:
We wont submit telemetry if the nested try/catch above throws. Is this intended/possible?
Consider wrapping this function in telemtry.run instead of emitting once at the end.

Reply:
I think it would make more sense to move all of the code in this catch block (except `await terminate` in the inner block) to the outer finally block.